esphome/components/modbus_controller/modbus_controller.cpp

```diff
@@ -22,7 +22,7 @@ bool ModbusController::send_next_command_() {
     auto &command = command_queue_.front();
 
     // remove from queue if command was sent too often
-    if (command->send_countdown < 1) {
+    if (command->send_counter >= this->max_cmd_retries_) {
```

Comment:
```suggestion
    if (command->send_counter == this->max_cmd_retries_) {
```
Can't ever be greater. But this logic could be moved into the `ModbusCommandItem` class so `send_counter` becomes protected, and gets modified only within the class. Either that or move the increment of the counter into this function - the logic should be confined to one class to make it easier to understand and validate.